File: MyFavsDB/lib/src/models/game_search_result.dart
class GameSearchResult {
  final int id;
  final String name;
  final String released;
  final String backgroundImage;
  final double rating;
  final int ratingTop;
  final int metacritic;
  final String description;
  final List<GameGenre> genres;
  final List<GamePlatform> platforms;
  final List<GamePublisher> publishers;
  final List<GameDeveloper> developers;

  GameSearchResult({
    required this.id,
    required this.name,
    required this.released,
    required this.backgroundImage,
    required this.rating,
    required this.ratingTop,
    required this.metacritic,
    required this.description,
    required this.genres,
    required this.platforms,
    required this.publishers,
    required this.developers,
  });

  factory GameSearchResult.fromMap(Map<String, dynamic> map) {
    return GameSearchResult(
      id: map['id'] ?? 0,
      name: map['name'] ?? '',
      released: map['released'] ?? '',
      backgroundImage: map['background_image'] ?? '',
      rating: (map['rating'] ?? 0.0).toDouble(),
      ratingTop: map['rating_top'] ?? 0,
      metacritic: map['metacritic'] ?? 0,
      description: map['description'] ?? '',
      genres: (map['genres'] as List<dynamic>?)
          ?.map((genre) => GameGenre.fromMap(genre))
          .toList() ?? [],
      platforms: (map['platforms'] as List<dynamic>?)
          ?.map((platform) => GamePlatform.fromMap(platform))
          .toList() ?? [],
      publishers: (map['publishers'] as List<dynamic>?)
          ?.map((publisher) => GamePublisher.fromMap(publisher))
          .toList() ?? [],
      developers: (map['developers'] as List<dynamic>?)
          ?.map((developer) => GameDeveloper.fromMap(developer))
          .toList() ?? [],
    );
  }

  String get displayRating => rating > 0 ? rating.toStringAsFixed(1) : 'N/A';
  String get displayMetacritic => metacritic > 0 ? metacritic.toString() : 'N/A';
  String get displayGenres => genres.take(3).map((g) => g.name).join(', ');
  String get displayPlatforms => platforms.take(3).map((p) => p.platform.name).join(', ');
}

class GameGenre {
  final int id;
  final String name;

  GameGenre({required this.id, required this.name});

  factory GameGenre.fromMap(Map<String, dynamic> map) {
    return GameGenre(
      id: map['id'] ?? 0,
      name: map['name'] ?? '',
    );
  }
}

class GamePlatform {
  final PlatformInfo platform;

  GamePlatform({required this.platform});

  factory GamePlatform.fromMap(Map<String, dynamic> map) {
    return GamePlatform(
      platform: PlatformInfo.fromMap(map['platform'] ?? {}),
    );
  }
}

class PlatformInfo {
  final int id;
  final String name;

  PlatformInfo({required this.id, required this.name});

  factory PlatformInfo.fromMap(Map<String, dynamic> map) {
    return PlatformInfo(
      id: map['id'] ?? 0,
      name: map['name'] ?? '',
    );
  }
}

class GamePublisher {
  final int id;
  final String name;

  GamePublisher({required this.id, required this.name});

  factory GamePublisher.fromMap(Map<String, dynamic> map) {
    return GamePublisher(
      id: map['id'] ?? 0,
      name: map['name'] ?? '',
    );
  }
}

class GameDeveloper {
  final int id;
  final String name;

  GameDeveloper({required this.id, required this.name});

  factory GameDeveloper.fromMap(Map<String, dynamic> map) {
    return GameDeveloper(
      id: map['id'] ?? 0,
      name: map['name'] ?? '',
    );
  }
} 

File: backend/src/main/java/com/myfavsdb/dto/GameSearchResult.java
package com.myfavsdb.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

public class GameSearchResult {
    @JsonProperty("id") public Long id;
    @JsonProperty("name") public String name;
    @JsonProperty("released") public String released;
    @JsonProperty("background_image") public String backgroundImage;
    @JsonProperty("rating") public Double rating;
    @JsonProperty("rating_top") public Integer ratingTop;
    @JsonProperty("metacritic") public Integer metacritic;
    @JsonProperty("description") public String description;
    @JsonProperty("genres") public List<GameGenre> genres;
    @JsonProperty("platforms") public List<GamePlatform> platforms;
    @JsonProperty("publishers") public List<GamePublisher> publishers;
    @JsonProperty("developers") public List<GameDeveloper> developers;
    @JsonProperty("short_screenshots") public List<GameScreenshot> shortScreenshots;

    public String getFullImageUrl() {
        if (backgroundImage != null && !backgroundImage.isEmpty()) {
            return backgroundImage;
        }
        return null;
    }

    public String getDisplayRating() {
        if (rating != null) {
            return String.format("%.1f", rating);
        }
        return "N/A";
    }

    public String getDisplayMetacritic() {
        if (metacritic != null) {
            return metacritic.toString();
        }
        return "N/A";
    }

    public String getDisplayGenres() {
        if (genres != null && !genres.isEmpty()) {
            return genres.stream()
                .map(genre -> genre.name)
                .limit(3)
                .reduce((a, b) -> a + ", " + b)
                .orElse("");
        }
        return "";
    }

    public String getDisplayPlatforms() {
        if (platforms != null && !platforms.isEmpty()) {
            return platforms.stream()
                .map(platform -> platform.platform.name)
                .limit(3)
                .reduce((a, b) -> a + ", " + b)
                .orElse("");
        }
        return "";
    }

    public static class GameGenre {
        @JsonProperty("id") public Long id;
        @JsonProperty("name") public String name;
    }

    public static class GamePlatform {
        @JsonProperty("platform") public PlatformInfo platform;
    }

    public static class PlatformInfo {
        @JsonProperty("id") public Long id;
        @JsonProperty("name") public String name;
    }

    public static class GamePublisher {
        @JsonProperty("id") public Long id;
        @JsonProperty("name") public String name;
    }

    public static class GameDeveloper {
        @JsonProperty("id") public Long id;
        @JsonProperty("name") public String name;
    }

    public static class GameScreenshot {
        @JsonProperty("id") public Long id;
        @JsonProperty("image") public String image;
    }
} 